### PR TITLE
[Citadel] Citadel support

### PR DESCRIPTION
**Pull request by chapulina**
* Ignore `ros_ign_point_cloud` until #40 is resolved
* Accept both ign-gazebo 2 and ign-gazebo 3 for the demos
* Other packages not affected, since we're still using `msgs4` and `transport7`

### CODE REVIEW

File: ros_ign_gazebo_demos/README.md [hunk 1]
```diff
@@ -55,7 +55,7 @@ Then send a command
 Depth camera data can be obtained as:
 
 * `sensor_msgs/Image`, through `ros_ign_bridge` or `ros_ign_image`
-* `sendor_msgs/PointCloud2`, through `ros_ign_point_cloud`
+* `sendor_msgs/PointCloud2`, through `ros_ign_point_cloud` (See issue #40)
```
Comment (JShep1) [inline ros_ign_gazebo_demos/README.md]: Did you mean `sensor` here?

File: ros_ign_gazebo_demos/README.md [hunk 2]
```diff
@@ -72,7 +72,7 @@ Using Ignition Gazebo plugin:
 GPU lidar data can be obtained as:
 
 * `sensor_msgs/LaserScan`, through the `ros_ign_bridge`
-* `sendor_msgs/PointCloud2`, through the `ros_ign_bridge` or `ros_ign_point_cloud`
+* `sendor_msgs/PointCloud2`, through the `ros_ign_bridge` or `ros_ign_point_cloud` (See issue #40)
```
Comment (JShep1) [inline ros_ign_gazebo_demos/README.md]: Same as above

File: ros_ign_gazebo_demos/README.md [hunk 3]
```diff
@@ -105,7 +105,7 @@ Publishes magnetic field readings.
 RGBD camera data can be obtained as:
 
 * `sensor_msgs/Image`, through `ros_ign_bridge` or `ros_ign_image`
-* `sendor_msgs/PointCloud2`, through `ros_ign_bridge` or `ros_ign_point_cloud`
+* `sendor_msgs/PointCloud2`, through `ros_ign_bridge` or `ros_ign_point_cloud` (See issue #40)
```
Comment (JShep1) [inline ros_ign_gazebo_demos/README.md]: Same as above

**Review by JShep1 — CHANGES_REQUESTED** [review]

File: ros_ign_bridge/CMakeLists.txt [hunk 4]
```diff
@@ -18,11 +18,25 @@ find_package(catkin REQUIRED COMPONENTS
...
+
+# Blueprint
+else()
+  find_package(ignition-transport7 REQUIRED)
+  if (NOT ignition-transport7_FOUND)
+    BUILD_ERROR ("Missing: Ignition Transport version 7 or 8")
+  endif()
+  set(IGN_TRANSPORT_VER ${ignition-transport7_VERSION_MAJOR})
+
+  find_package(ignition-msggs4 REQUIRED)
```
Comment (JShep1) [inline ros_ign_bridge/CMakeLists.txt]: msggs -> msgs

File: ros_ign_bridge/CMakeLists.txt [hunk 5]
```diff
@@ -18,11 +18,25 @@ find_package(catkin REQUIRED COMPONENTS
                sensor_msgs
                std_msgs)
 
-find_package(ignition-msgs4 QUIET REQUIRED)
-set(IGN_MSGS_VER ${ignition-msgs4_VERSION_MAJOR})
-
-find_package(ignition-transport7 QUIET REQUIRED)
-set(IGN_TRANSPORT_VER ${ignition-transport7_VERSION_MAJOR})
+# Citadel
+find_package(ignition-transport8 REQUIRED)
```
Comment (JShep1) [inline ros_ign_bridge/CMakeLists.txt]: `REQUIRED` -> `QUIET`

File: ros_ign_image/CMakeLists.txt [hunk 6]
```diff
@@ -16,11 +16,25 @@ find_package(catkin REQUIRED COMPONENTS
   roscpp
   sensor_msgs)
 
-find_package(ignition-msgs4 QUIET REQUIRED)
-set(IGN_MSGS_VER ${ignition-msgs4_VERSION_MAJOR})
+# Citadel
+find_package(ignition-transport8 REQUIRED)
```
Comment (JShep1) [inline ros_ign_image/CMakeLists.txt]: `REQUIRED` -> `QUIET`

File: ros_ign_image/CMakeLists.txt [hunk 7]
```diff
@@ -16,11 +16,25 @@ find_package(catkin REQUIRED COMPONENTS
...
+
+# Blueprint
+else()
+  find_package(ignition-transport7 REQUIRED)
+  if (NOT ignition-transport7_FOUND)
+    BUILD_ERROR ("Missing: Ignition Transport version 7 or 8")
+  endif()
+  set(IGN_TRANSPORT_VER ${ignition-transport7_VERSION_MAJOR})
+
+  find_package(ignition-msggs4 REQUIRED)
```
Comment (JShep1) [inline ros_ign_image/CMakeLists.txt]: msggs -> msgs

**Review by JShep1 — CHANGES_REQUESTED** [review]

**Review by claireyywang — APPROVED** [review]
LGTM other than the typos.

File: ros_ign_bridge/CMakeLists.txt [hunk 8]
```diff
@@ -18,11 +18,25 @@ find_package(catkin REQUIRED COMPONENTS
...
+if (ignition-transport8_FOUND)
+  set(IGN_TRANSPORT_VER ${ignition-transport8_VERSION_MAJOR})
+
+  find_package(ignition-msgs5 REQUIRED)
+  set(IGN_MSGS_VER ${ignition-msgs5_VERSION_MAJOR})
+
+# Blueprint
+else()
+  find_package(ignition-transport7 REQUIRED)
+  if (NOT ignition-transport7_FOUND)
```
Comment (JShep1) [inline ros_ign_bridge/CMakeLists.txt]: This if statement can be deleted

File: ros_ign_image/CMakeLists.txt [hunk 9]
```diff
@@ -16,11 +16,25 @@ find_package(catkin REQUIRED COMPONENTS
...
 
-find_package(ignition-transport7 QUIET REQUIRED)
-set(IGN_TRANSPORT_VER ${ignition-transport7_VERSION_MAJOR})
+  find_package(ignition-msgs5 REQUIRED)
+  set(IGN_MSGS_VER ${ignition-msgs5_VERSION_MAJOR})
+
+# Blueprint
+else()
+  find_package(ignition-transport7 REQUIRED)
+  if (NOT ignition-transport7_FOUND)
```
Comment (JShep1) [inline ros_ign_image/CMakeLists.txt]: This if statement can be deleted

**Review by JShep1 — CHANGES_REQUESTED** [review]
LGTM! Builds cleanly with `Citadel` and `Blueprint` with all demos working once the suggested changes are made.

**Review by JShep1 — APPROVED** [review]
LGTM

**Pull request merged**